x-pack/plugins/task_manager/server/metrics/task_run_metrics_aggregator.ts

```diff
@@ -9,70 +9,79 @@ import { JsonObject } from '@kbn/utility-types';
 import { isOk, unwrap } from '../lib/result_type';
 import { TaskLifecycleEvent } from '../polling_lifecycle';
 import { ErroredTask, RanTask, TaskRun } from '../task_events';
-import { SuccessRate, SuccessRateCounter } from './success_rate_counter';
+import { MetricCounterService } from './counter/metric_counter_service';
 import { ITaskMetricsAggregator } from './types';
 
 const taskTypeGrouping = new Set<string>(['alerting:', 'actions:']);
 
+enum TaskRunKeys {
+  SUCCESS = 'success',
+  RUN_WITHIN_TIMEOUT = 'on_time',
```

Comment:
Open to alternative naming suggestions!

Reply:
@kobelb Any feedback as to naming? I'm keeping track of the number of tasks that run within the timeout. `on_time` sounds a little awkward but I couldn't come up with a good alternative

Reply:
Naive question, will the "success" bucket be different than the "on time" bucket?

Reply:
It might be! Because task cancellation is an optional function, a task could be cancelled due to timeout but not actually stop running. In this case, they would complete successfully, incrementing the success count but be considered timed out.

Reply:
After discussion, renaming to `not_timed_out`